clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3650,6 +3650,14 @@ void CompilerInvocationBase::GenerateLangArgs(const LangOptions &Opts,
       GenerateArg(Consumer, OPT_ftrigraphs);
   }
 
+  if (T.isOSzOS()) {
+    if (!Opts.ZOSExt)
+      GenerateArg(Consumer, OPT_fno_zos_extensions);
+  } else {
+    if (Opts.ZOSExt)
+      GenerateArg(Consumer, OPT_fzos_extensions);
+  }
```

Comment:
I think this can be simplified since zos-extensions are on by default on z/OS.

Reply:
Thanks, I refactored the code a bit